ls1intum/Artemis
# `Programming exercises`: Fix export of multiple repositories

### Checklist
#### General



#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
Tutors could not export multiple student repositories since the "Export All Repositories" Flag was enforced during export. but this flag is only allowed for instructors which resulted in a 403 response of the server. 
However, this also led to a misleading UI for instructors, since regardless what they chose for this flag, it was used anyways. So for them it was also not possible to export only the repositories of a subset of students. 

### Description
This PR moves the logic for the allRepositories flag to the init function so that it doesn't destroy the UI for instructors.
Also, this PR adds warnings, if there are no submissions for an exercise or a non-existing student was specified.
Lastly, the disabled Button for invalid export form was fixed, so it is no longer possible to submit without students specified or without allRepositories flag.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor
- 1 Programming Exercise with Submissions

1. Log in to Artemis as instructor
2. Navigate to Course Administration -> Exercise Overview
3. Select Multiple Exercises
4. Click "Download Repositories"
5. Verify that the export works
6. Select single exercise
7. Export once with allStudents flag and once with not-existing student identifier
8. Verify that the first one works and the second one displays a warning
9. Log in as tutor
10. Go to scores view of an exercise
11. specify multiple student names and export the repositories
12. Verify that it works


### Review Progress






#### Code Review
#### Manual Tests


### Test Coverage







### Screenshots
![image](https://github.com/ls1intum/Artemis/assets/78978542/c2c81ffc-a238-41fe-b586-e3f5aec27e3a)
